src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -247,56 +342,74 @@ public SNIHandle CreateConnectionHandle(object callbackObject, string fullServer
         /// <param name="callbackObject">Asynchronous I/O callback object</param>
         /// <param name="parallel">Should MultiSubnetFailover be used</param>
         /// <returns>SNITCPHandle</returns>
-        private SNITCPHandle CreateTcpHandle(string fullServerName, long timerExpire, object callbackObject, bool parallel)
+        private SNITCPHandle CreateTcpHandle(string fullServerName, long timerExpire, object callbackObject, bool parallel, ref byte[] spnBuffer)
         {
             // TCP Format: 
             // tcp:<host name>\<instance name>
             // tcp:<host name>,<TCP/IP port number>
 
-            if (string.IsNullOrWhiteSpace(fullServerName))
+            string fqdnHostName = null; ;
+            int port = -1;
+            Exception exception = null;
+
+            if (string.IsNullOrWhiteSpace(fullServerName)) // when fullServerName is empty
             {
-                int defaultInstancePort = TryToGetDefaultInstancePort(DefaultHostname);
-                return new SNITCPHandle(DefaultHostname, (defaultInstancePort > 0 ? defaultInstancePort : DefaultSqlServerPort), timerExpire, callbackObject, parallel);
+                fqdnHostName = GetFullyQualifiedDomainName(DefaultHostName);
+                port = DefaultSqlServerPort;
+                spnBuffer = GetMsSqlServerSPN(fqdnHostName, port);
```

Comment:
The SPN shouldn't be created for all Tcp Connections. Check for Integrated Security in the connection string to populate the SPN.
There are test failure in CI because of failure to create SPN.